### ISSUE TITLE
docker images spam question (issue?)

### ISSUE THREAD

**Issue by AaronDowdy** [testcontainers/testcontainers-node#189]
Really great node package! But I'm seeing a quirky behavior regarding leftover images after I run temporary containers.

I use the following to create throwaway containers based on a path and a Dockerfile:

```
const myGenericContainerbuilder = GenericContainer.fromDockerfile(buildContextPath, dockerfile);
...
let myContainer = myGenericContainerbuilder.build();
...
await myContainer.start()
...
// do my work
...
await myContainer.stop()
```

This works and creates the image, which in turn I can use to instantiate a container. The containers correctly appear to clean themselves up (via Ryuk I believe).

However, even running `docker image prune` commands, I'm unable to clean up temorary images, and end up spammed when I run `docker images`:
![Screen Shot 2021-02-16 at 3 45 05 PM](https://user-images.githubusercontent.com/60189338/108131064-f3440700-706d-11eb-8b32-15983b9ac930.png)

Seemingly similar looking issue, though I'm not seeing this related to logs (seems the conversation is confused between images and containers -- containers are cleaned up properly, but not images).
https://github.com/testcontainers/testcontainers-node/issues/69

in case it matters, I'm running testcontainers v 6.11,  node 12.0.0 on macOS Big Sur 1.11, using the Docker desktop client version 3.10 for macOS

EDIT - to be clear, I'm pretty sure I could run `docker image prune -a` but that deletes a bunch of other stuff I want to not redownload. Bad experience there

**Comment by cristianrgreco** [comment]
Thanks for raising @AaronDowdy, yes it seems the images aren't being cleaned up as expected by ryuk. I've asked @bsideup over on Slack for some help as my understanding is that they should be being cleaned up

**Comment by everflux** [comment]
I observed this behavior as well. It might be caused by aborting the tests due to a failure.
It would be really helpful if the images would have at least a label, so I could delete by label=testcontainers or something like that

**Comment by everflux** [comment]
I indeed missed that! Thanks, this helps indeed.
The command I actually used is 

```
docker image prune -a -f --filter 'label=org.testcontainers.session-id' 
```
The issue occurs for me when executing tests and the building of an image fails.

```
Error Error: Failed to build image
    at GenericContainerBuilder.<anonymous> (/.../integrationstests/node_modules/testcontainers/dist/generic-container.js:48:23)
    at Generator.next (<anonymous>)
    at fulfilled (/.../integrationstests/node_modules/testcontainers/dist/generic-container.js:5:58)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
Ending test run due to Error
```

**Comment by cristianrgreco** [comment]
Thanks @everflux, I have updated my comment to show the correct command so others don't make the same mistake.

**Comment by cristianrgreco** [comment]
The issue is that when we build an image, we give it a name and a tag, the result is that the image is not "dangling". Because of this, `docker image prune` will not clear them unless you specify `-a` "Remove all unused images, not just dangling ones". I've raised a PR for moby-ryuk which implements this behaviour, I've tested the change and it works as expected. Just waiting for that PR to be merged/released for this issue to be resolved.
